yt_dlp/extractor/common.py

```diff
@@ -2035,10 +2035,11 @@ def _parse_m3u8_formats_and_subtitles(
             video_id=None):
         formats, subtitles = [], {}
 
-        if '#EXT-X-FAXS-CM:' in m3u8_doc:  # Adobe Flash Access
-            return formats, subtitles
-
-        has_drm = re.search(r'#EXT-X-(?:SESSION-)?KEY:.*?URI="skd://', m3u8_doc)
+        # NOTE: '#' needs to be escaped since it's a comment character in multiline regex ('\#' doesn't work)
+        has_drm = re.search(r'''(?x)(?:
+                            [#]EXT-X-FAXS-CM:|                        # Adobe Flash Access
+                            [#]EXT-X-(?:SESSION-)?KEY:.*?URI="skd://  # Apple FairPlay
+                            )''', m3u8_doc)
```

Comment:
```suggestion
        has_drm = re.search('|'.join([
            r'#EXT-X-FAXS-CM:',  # Adobe Flash Access
            r'#EXT-X-(?:SESSION-)?KEY:.*?URI="skd://',  # Apple FairPlay
        ], m3u8_doc)
```
is far more readable imo. What do u think?